src/common/src/catalog/physical_table.rs

```diff
@@ -18,12 +18,16 @@ use super::{ColumnDesc, OrderedColumnDesc, TableId};
 /// information for compute node to access data of the table.
 #[derive(Debug, Clone, Default)]
 pub struct TableDesc {
-    /// id of the table, to find in Storage()
+    /// Id of the table, to find in storage.
     pub table_id: TableId,
-    /// the primary key columns' descriptor
-    pub pk: Vec<OrderedColumnDesc>,
-    /// all columns in the table, noticed it is NOT sorted by columnId in the vec
+    /// The keys used to sort in storage.
+    pub order_desc: Vec<OrderedColumnDesc>,
```

Comment:
`order_desc` also has a full description of Column, which is duplicated with `columns`. Perhaps `FieldOrder` is better here?

```
pub struct FieldOrder {
    pub index: usize,
    pub direct: Direction,
}
```

Reply:
Agree. Maybe we can do the refactor and de-dup along with https://github.com/singularity-data/risingwave/issues/2145. Currently I saw `columns` the only source of truth, and `order_desc` is never used to get column info.